xtask/src/commands/mod.rs

```diff
@@ -248,26 +235,56 @@ pub fn tests(workspace: &Path, args: TestsArgs, action: CargoAction) -> Result<(
 
     let mut commands = CargoCommandBatcher::new();
     // Execute the specified action:
-    if tests.iter().any(|test| test.matches(test_arg.as_deref())) {
-        for test in tests
-            .iter()
-            .filter(|test| test.matches(test_arg.as_deref()))
+    // If user sets some specific test(s)
+    if let Some(test_arg) = args.test.as_deref() {
```

Comment:
Nitpick (mostly useless): I think that a user could do `.... --tests gpio, gpio::async_edge` and it will ran twice if im not mistaken.